### PR TITLE
Actions

### PR DESCRIPTION
**Pull request by jburel**
* Remove usage of travis
* replace by actions

This is the approach that we could move the Maven template to publish to github

cc @joshmoore @sbesson

### CODE REVIEW

**Comment by jburel** [comment]
tested via https://github.com/jburel/omero-downloader/releases/tag/v0.2.5

**Comment by jburel** [comment]
As discussed today, this could be added to the maven/gradle template when publishing artifacts on GH.

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        with:
+          name: artifacts
+          path: ${{ env.artifact_dir }}
+          if-no-files-found: error
+  release:
+    runs-on: ubuntu-latest
+    needs: build
+    if: startsWith(github.ref, 'refs/tags')
+    env:
+      publishing_target: github # default is github
```
Comment (sbesson) [inline .github/workflows/maven.yml]: From the template perspective, are you foreseeing jobs publishing to multiple targets? If so, will this need to become an array?

**Review by sbesson — COMMENTED** [review]
Functionally works as expected.

One question about the artifact/release lifecyle especially if this gets added to the template:

- the upload-artifact action bundle the entire content of `target/*` including JARs, binaries but also classes, tests etc 
- the upload-release-action only seems to be converting the files directly under `target` as release assets.

Is that the correct expectation/contract for these repositories? Are we happy with having the the artifacts always uploaded across repositories?

**Comment by jburel** [comment]
* This is a mistake in my setup, only the artifacts we wish to release should be uploaded. 
I will push a commit to fix that.
Following the discussion yesterday, for an atomic release the artifacts must be uploaded.
I will add a flag ``upload_artifact`` defaulting to true, so one can adjust that

**Comment by jburel** [comment]
In the template, it will be necessary to indicate what needs to be uploaded. It will need to be adjusted on a case-by-case basis.
For downloader, target contains many things, other like insight contains the artifacts to publish.
I will add doc to the template.

**Comment by jburel** [comment]
New logic tested via run https://github.com/jburel/omero-downloader/actions/runs/401135241

**Comment by jburel** [comment]
Using matrix for publishing https://github.com/jburel/omero-downloader/actions/runs/401231272

**Review by sbesson — APPROVED** [review]
I think this matches the discussion we had a few days ago around the building and uploading of release assets.

Merging and we can refine as we cut new releases of `omero-downloader`.

**Pull request merged**